src/core/electrostatics_magnetostatics/p3m-common.hpp

```diff
@@ -21,41 +21,55 @@
 #ifndef _P3M_COMMON_H
 #define _P3M_COMMON_H
 /** \file
- *  common functions for dipolar and charge p3m.
+ *  Common functions for dipolar and charge P3M.
  *
  *  We use here a P3M (Particle-Particle Particle-Mesh) method based
  *  on the Ewald summation. Details of the used method can be found in
  *  Hockney/Eastwood and Deserno/Holm. The file p3m contains only the
  *  Particle-Mesh part.
  *
  *  Further reading:
- *  <ul>
- *  <li> P.P. Ewald,
- *       <i>Die Berechnung optischer und elektrostatischer Gitterpotentiale</i>,
+ *  -    P. P. Ewald,
+ *       *Die Berechnung optischer und elektrostatischer Gitterpotentiale*,
  *       Ann. Phys. (64) 253-287, 1921
- *  <li> R. W. Hockney and J. W. Eastwood,
- *       <i>Computer Simulation Using Particles</i>,
+ *  -    R. W. Hockney and J. W. Eastwood,
+ *       *Computer simulation using particles*,
  *       IOP, London, 1988
- *  <li> M. Deserno and C. Holm,
- *       <i>How to mesh up {E}wald sums. I. + II.</i>,
+ *  -    M. Deserno and C. Holm,
+ *       *How to mesh up Ewald sums I + II*,
  *       J. Chem. Phys. (109) 7678, 1998; (109) 7694, 1998
- *  <li> M. Deserno, C. Holm and H. J. Limbach,
- *       <i>How to mesh up {E}wald sums. </i>,
+ *  -    M. Deserno, C. Holm and H. J. Limbach,
+ *       *How to mesh up Ewald sums*,
  *       in Molecular Dynamics on Parallel Computers,
  *       Ed. R. Esser et al., World Scientific, Singapore, 2000
- *  <li> M. Deserno,
- *       <i>Counterion condensation for rigid linear polyelectrolytes</i>,
- *       PhdThesis, Universit{\"a}t Mainz, 2000
- *  <li> J.J. Cerda, P3M for dipolar interactions. J. Chem. Phys, 129, xxx
- * ,(2008).
- *  </ul>
+ *  -    M. Deserno,
+ *       *Counterion condensation for rigid linear polyelectrolytes*,
+ *       PhD Thesis, Universität Mainz, 2000
+ *  -    J. J. Cerda,
+ *       *P3M for dipolar interactions*,
+ *       J. Chem. Phys (129) 234104, 2008
  *
  */
 #include "config.hpp"
 #include "utils.hpp"
 
 #if defined(P3M) || defined(DP3M)
 
+/** Error Codes for p3m tuning (version 2) */
+enum P3M_TUNE_ERROR : long {
```

Comment:
Why do explicitly state the underlying type here? And why `long`?

Reply:
```suggestion
enum P3M_TUNE_ERROR {
```
The `long` brings nothing, actually. My mistake. The underlying type is deduced automatically based on the largest value in the enum.